src/main/java/yalter/mousetweaks/loaders/MouseTweaksForge.java

```diff
@@ -1,24 +1,47 @@
 package yalter.mousetweaks.loaders;
 
+import com.gtnewhorizon.gtnhlib.config.ConfigException;
+import com.gtnewhorizon.gtnhlib.config.ConfigurationManager;
+
 import cpw.mods.fml.common.FMLCommonHandler;
 import cpw.mods.fml.common.Mod;
 import cpw.mods.fml.common.event.FMLInitializationEvent;
+import cpw.mods.fml.common.event.FMLPreInitializationEvent;
 import cpw.mods.fml.common.eventhandler.SubscribeEvent;
 import cpw.mods.fml.common.gameevent.TickEvent;
+import yalter.mousetweaks.Config;
 import yalter.mousetweaks.Constants;
 import yalter.mousetweaks.Main;
+import yalter.mousetweaks.config.MTConfig;
 
 @Mod(
         modid = Constants.MOD_ID,
         name = Constants.MOD_NAME,
         version = Constants.MOD_VERSION,
+        guiFactory = "yalter.mousetweaks.config.MTGuiConfigFactory",
         acceptedMinecraftVersions = "[1.7.10]",
         acceptableRemoteVersions = "*")
```

Comment:
Please set GTNHLib (with min. version) as required dependency.

Reply:
whoops, forgot that

Reply:
I tested if this actually works because [ConfigurationManager](https://github.com/GTNewHorizons/MouseTweaks/pull/5/files#diff-f2a21e6cfd73956296c32f3dfa667bfbc59c1cdd0f6bdf50d68332d754f3c542R35-R36) and [ConfigException](https://github.com/GTNewHorizons/MouseTweaks/pull/5/files#diff-f2a21e6cfd73956296c32f3dfa667bfbc59c1cdd0f6bdf50d68332d754f3c542R37) is referenced in this class and I wasn't sure if it would throw a ClassNotFoundException. Turns out, it does:
<details>
<summary>crash-2024-01-13_16.34.22-client.txt</summary>

```
---- Minecraft Crash Report ----
// Shall we play a game?

Time: 1/13/24 4:34 PM
Description: There was a severe problem during mod loading that has caused the game to fail

cpw.mods.fml.common.LoaderException: java.lang.NoClassDefFoundError: com/gtnewhorizon/gtnhlib/config/ConfigException
	at cpw.mods.fml.common.LoadController.transition(LoadController.java:163)
	at cpw.mods.fml.common.Loader.loadMods(Loader.java:544)
	at cpw.mods.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:208)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:480)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:878)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
Caused by: java.lang.NoClassDefFoundError: com/gtnewhorizon/gtnhlib/config/ConfigException
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at cpw.mods.fml.common.FMLModContainer.constructMod(FMLModContainer.java:440)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212)
	at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119)
	at cpw.mods.fml.common.Loader.loadMods(Loader.java:513)
	... 13 more
Caused by: java.lang.ClassNotFoundException: com.gtnewhorizon.gtnhlib.config.ConfigException
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 38 more
Caused by: java.lang.NullPointerException
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:182)
	... 40 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.7.10
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_292, AdoptOpenJDK
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), AdoptOpenJDK
	Memory: 819841424 bytes (781 MB) / 1149239296 bytes (1096 MB) up to 3817865216 bytes (3641 MB)
	JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms512m -Xmx4096m
	AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP v9.05 FML v7.10.99.99 Minecraft Forge 10.13.4.1614 8 mods loaded, 8 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
	UC	mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) 
	UC	FML{7.10.99.99} [Forge Mod Loader] (forge-1.7.10-10.13.4.1614-1.7.10-universal.jar) 
	UC	Forge{10.13.4.1614} [Minecraft Forge] (forge-1.7.10-10.13.4.1614-1.7.10-universal.jar) 
	UC	CodeChickenCore{1.1.13} [CodeChicken Core] (minecraft.jar) 
	UC	NotEnoughItems{2.4.13-GTNH} [NotEnoughItems] (NotEnoughItems-2.4.13-GTNH.jar) 
	UC	BiomesOPlenty{2.1.0} [Biomes O' Plenty] (BiomesOPlenty-1.7.10-2.1.0.2308-universal.jar) 
	UC	ExtrabiomesXL{3.16.4} [ExtrabiomesXL] (extrabiomesxl_1.7.10-3.16.4.jar) 
	UE	MouseTweaks{2.4.9-GTNH-newConfig.5+e65a4ff31c} [Mouse Tweaks] (MouseTweaks-2.4.9-GTNH-newConfig.5+e65a4ff31c.jar) 
	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.6.0 NVIDIA 536.23' Renderer: 'NVIDIA GeForce GTX 970/PCIe/SSE2'
```

</details>

However, the problem is *not* that these two classes are referenced in `preInit`. The problem is that [`"useDependencyInformation": true`](https://github.com/GTNewHorizons/MouseTweaks/blob/master/src/main/resources/mcmod.info#L19) makes FML ignore the dependencies declared in the `@Mod` annotation and uses the dependencies declared in `mcmod.info` instead when checking for required mods/versions.
To solves this, either set `useDependencyInformation` to `false` or move the dependency declaration to the `mcmod.info` file.

Reply:
Removed from the mcmod file (default is false)